### DISCUSSION TITLE
Call Interceptors / Middleware before Mutation is executed

### DISCUSSION THREAD

**Discussion by msravi16** [hasura/graphql-engine#8351] [Question]
Hi Team,

I am new to Hasura, I am impressed by the instant GraphQL and Security features.
We have many mutations which are allready in production. 
There is a new requirment where I want to change the request payload before I store it in database.
I understood the concept of Actions, if I am developing new API then I will create Action and call mutation from the action service.
But, in my case there are many clients who are using mutation so, if I create new action then there will change in client side code also. I dont want to disturb client side application.

Is there any way where I call middleware/service to change payload before it reaches postgres.

Thanks,
Ravi

**Comment by cemremengu** [comment]
I am also looking to add filters to a query before it reaches postgres
